core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -620,6 +620,14 @@ private[spark] class TaskSetManager(
     // Note: "result.value()" only deserializes the value when it's called at the first time, so
     // here "result.value()" just returns the value and won't block other threads.
     sched.dagScheduler.taskEnded(tasks(index), Success, result.value(), result.accumUpdates, info)
+    // Kill other task attempts if any as the one attempt succeeded
+    for (attemptInfo <- taskAttempts(index) if attemptInfo.attemptNumber != info.attemptNumber
```

Comment:
I think you don't need the middle condition here (if attemptInfo.attemptNumber != info.attemptNumber) since this attempt will no longer be running (since markSuccessful() was called above), so the last condition will fail?